test/code/providers/runas_provider/runasprovider_test.cpp

```diff
@@ -101,6 +103,17 @@ class SCXRunAsProviderTest : public CPPUNIT_NS::TestFixture
             configurator->SetAllowRoot(true);
         }
         SCXCore::g_RunAsProvider.SetConfigurator(configurator);
+        
+        int err = system("ln -s /tmp ./tmpdir");
+        if(err) 
+        { 
+        	std::ostringstream os;
+            os << "Unexpected error in setUp: " << SCXCoreLib::strerror(err & 127);
+            std::cout<<os.str()<<std::endl;
+        }
+        SCXCoreLib::SCXHandle<SCXSystemLib::SystemInfo> si(new SCXSystemLib::SystemInfo());
+        si->SetDefaultTemporaryDirectory(L"./tmpdir/");
+        SCXCore::g_RunAsProvider.SetSystemInfo(si);
```

Comment:
Rather than creating with system(), I suggest you use SCXCoreLib::SCXProcess::Run(). This allows you to capture stderr/stdout and print that in a controlled way in the failure message.

Not sure that it makes sense to create this for every test (even though just a few tests need it). You can create it in the test(s) that need it, then either leave the code in tearDown() (so you're sure it's gone), or set up a destructor to delete the link so you're certain it's gone. I prefer the later (it's limited to specific tests), but will take either to insure that the temporary directory is cleaned up, no matter what.

As commented elsewhere, no point of storing the temporary directory in SystemInfo. Just store it yourself in a string variable and be done with it.